frontend/app/views/spree/shared/_products.html.erb

```diff
@@ -24,8 +24,8 @@
         <div class="panel panel-default">
           <% cache(@taxon.present? ? [I18n.locale, current_currency, @taxon, product] : cache_key_for_product(product)) do %>
             <div class="panel-body text-center product-body">
-              <%= link_to small_image(product, itemprop: "image"), url, itemprop: 'url' %><br/>
-              <%= link_to truncate(product.name, length: 50), url, class: 'info', itemprop: "name", title: product.name %>
+              <%= link_to small_image(product, itemprop: "image") + content_tag(:span, truncate(product.name, length: 50), class: 'info', itemprop: "name", title: product.name), url, itemprop: 'url' %>
```

Comment:
@rehan-work This seems a bit complicated. Instead we can use a block with link_to and add image and span there.